### DISCUSSION TITLE
新编译了一个5.15.63版本的R68S固件，发现dnsmasq和adbyby-plus可能有兼容问题

### DISCUSSION THREAD

**Discussion by shzstone** [coolsnowwolf/lede#10047] [Q&A / 快速问答] [answered]
在新编译的固件里面，发现一启用adbyby-plus，dnsmasq就会自动退出，导致dns异常，无法上网，里面报的是04-dnsmasq.adblock这个文件的格式第一行不正确，禁用adbyby-plus，dnsmasq就会被重新自动拉起来，检查04-dnsmasq.adblock文件的内容，发现5.15.63版本里面的04-dnsmasq.adblock文件前面多了一些title，包含文件说明之类的，找了一台正常能工作的x86 LEDE也是5.15.63版本的，发现04-dnsmasq.adblock文件里面并没有这些说明类的信息。

正常的文件内容如下：

**Comment by shzstone** [comment]
root@OpenWrt:/# cat /var/etc/dnsmasq-adbyby.d/04-dnsmasq.adblock |head -10
address=/001adafg.xyz/0.0.0.0
address=/04jhgyt.com/0.0.0.0
address=/0577.la/0.0.0.0
address=/103.21.91.144/0.0.0.0
address=/1527ego.com/0.0.0.0
address=/15h3na0.xyz/0.0.0.0
address=/17chk.vip/0.0.0.0
address=/1dfsdfg.com/0.0.0.0
address=/1e4y54ve.cn/0.0.0.0
address=/1so8inqb.cn/0.0.0.0

**Comment by BG8JOA** [comment]
编译出的x86固件也有同样的问题

**Comment by shzstone** [comment]
新的规则文件多了以下内容，估计是不是主程序还没有来的及适配，导致的问题。。。
[Adblock Plus 2.0]
! Checksum: I6IQUu9cTq1Bv7+aSmlOVQ
! Version: 202209040621
! Title: EasyList China+EasyList
! Last modified: 04 Sep 2022 06:21 UTC
! Expires: 1 days (update frequency)
! Homepage: https://github.com/easylist/easylistchina/
!
! *** easylistchina.txt ***


而且文件格式也不一样了，老的规则文件格式为adress=xxx
address=/2ppgsmeh.cn/0.0.0.0
address=/32258228.com/0.0.0.0
address=/354664.top/0.0.0.0
address=/3dcn.me/0.0.0.0

新的规则文件则是：
-*-*-*-*.panda^$script,third-party
-1688-wp-media/ads/
-880-80-4.jpg
-90mh-gg.
.bid/ads/
.blog/ads/
.cam/ads/
.cc/ad/
.cc/js/ads/
.cc/s.php?id=$script,third-party

格式都不一样了。。

**Comment by shzstone** [comment, marked as answer]
今天git pull发现adbyby-plus配置文件有更新，重新编译更新了一下最新的版本，这个问题解决了。
